ci-operator/step-registry/ipi/deprovision/deprovision/ipi-deprovision-deprovision-commands.sh

```diff
@@ -11,6 +11,11 @@ export AWS_SHARED_CREDENTIALS_FILE=$cluster_profile/.awscred
 export AZURE_AUTH_LOCATION=$cluster_profile/osServicePrincipal.json
 
 echo "Deprovisioning cluster ..."
+if [ ! -f "${SHARED_DIR}/metadata.json"]; then
+  echo "Skipping: ${SHARED_DIR}/metadata.json not found."
+  return 0
+fi
```

Comment:
why are we running this step when we don't want it to do anything?

can you explain where this is becoming a problem?

Reply:
This step always gets ran. However, if there was an error prior to the metadata.json being created, such as during the config steps, then there isn't a cluster to be destroyed. This will remove the false-negative error from this pod when this is the case.

I had assumed it caused any following steps to be skipped. The upi-gcp-post step will after this step in the upi-gcp-post chain. However, I was wrong. So the only reason to follow through with this PR is to remove the false-negative error.

Reply:
Here is an example.

```
Deprovisioning cluster ... level=fatal msg="Failed while preparing to destroy cluster: open /tmp/installer/metadata.json: no such file or directory" error: failed to execute wrapped command: exit status 1
```

https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_releas/8233/rehearse-8233-pull-ci-openshift-installer-master-e2e-gcp-upi/9

Reply:
I think it makes sense for these to detect "nothing to do" and do nothing. Or just fail and not risk not doing something when you should.